runtime/lua/vim/lsp/buf.lua

```diff
@@ -634,7 +634,8 @@ function M.execute_command(command)
     command = { command.command, 's' },
     arguments = { command.arguments, 't', true }
   }
-  request('workspace/executeCommand', command)
+  command = { command=command.command, arguments=command.arguments }
```

Comment:
I don't think we use it yet,  but a `workDoneToken` property would also be allowed. 

https://microsoft.github.io/language-server-protocol/specification#workspace_executeCommand 
It extends WorkDoneProgressParams
https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#workDoneProgressParams

Reply:
I think part of the reason is that neovim by default just doesn't have the UI to display a (cancellable) long process.
That's why the tokens are currently in general not really used much.

If you've a concrete proposal, please open a new issue - it's something worth consideration.